src/wwwroot/controllers/LoginCtrl.js

```diff
@@ -5,65 +5,95 @@
     .module('dopplerRelay')
     .controller('LoginCtrl', LoginCtrl);
   LoginCtrl.$inject = [
-    '$scope',
     '$location',
     '$timeout',
     'auth',
-    '$translate'
+    '$translate',
+    'vcRecaptchaService'
   ];
 
-  function LoginCtrl($scope, $location, $timeout, auth, $translate) {
+  function LoginCtrl($location, $timeout, auth, $translate, vcRecaptchaService) {
 
-    $scope.forgotOpened = false;
-    $scope.forgotSuccessful = false;
+    var vm = this;
+    vm.forgotOpened = false;
+    vm.forgotSuccessful = false;
+    vm.updateValidation = updateValidation;
+    vm.gotoBottom = gotoBottom;
+    vm.submitLogin = submitLogin;
+    vm.removeSucceed = removeSucceed;
+    vm.submitForgot = submitForgot;
 
-    $scope.updateValidation = function () {
-      if ($scope.loginform && $scope.loginform.email) {
-        $scope.loginform.email.$setValidity('error', true);
+    function updateValidation(loginform) {
+      if (loginform && loginform.email) {
+        loginform.email.$setValidity('error', true);
       }
-      if ($scope.loginform && $scope.loginform.password) {
-        $scope.loginform.password.$setValidity('error', true);
+      if (loginform && loginform.password) {
+        loginform.password.$setValidity('error', true);
       }
     }
-    $scope.gotoBottom = function () {
-      $scope.forgotOpened = !$scope.forgotOpened;
+    function gotoBottom() {
+      vm.forgotOpened = !vm.forgotOpened;
     };
 
-    $scope.submitLogin = function () {
-      if ($scope.loginform.$invalid) {
+    function submitLogin (loginform) {
+      if (loginform.$invalid) {
         return;
       }
 
       var credentials = {
-        username: $scope.email,
-        password: $scope.password
+        username: loginform.email.$modelValue,
+        password: loginform.password.$modelValue
       };
 
       auth.login(credentials).then(function (result) {
         if (result.authenticated) {
           $location.path('/');
         } else {
-          $scope.loginform.email.$setValidity('error', false);
-          $scope.loginform.password.$setValidity('error', false);
+          loginform.email.$setValidity('error', false);
+          loginform.password.$setValidity('error', false);
         }
       });
     };
 
-    $scope.removeSucceed = function () {
-      if ($scope.forgotSuccessful) {
-        $scope.forgotSuccessful = false;
+    function removeSucceed(forgotEmail) {
+      if (vm.forgotSuccessful) {
+        vm.forgotSuccessful = false;
```

Comment:
_If_ is not necessary here.

Reply:
And why does it receives an email? and it is really necessary to call _removeSucceed_ from the view? could not the validation do it automatically?